crates/zksync/compiler/src/zksolc/compile.rs

```diff
@@ -319,22 +429,19 @@ impl ZkSolc {
                     .to_str()
                     .expect("Invalid Contract filename");
 
-                // Run Compiler (or use cached) and Handle Output
-                let artifact_paths =
-                    ZkSolcArtifactPaths::new(self.project.paths.artifacts.join(filename));
-
-                info!("Compiling {:?}...", contract_path);
-                let (output, contract_hash) = self.check_contract_is_cached(&contract_path)?;
-                let (output, maybe_artifact_paths) = match output {
-                    Some(output) => {
+                let artifacts = match self
+                    .check_cache(&self.project.paths.artifacts, &contract_path)?
+                {
+                    CachedContractEntry::Found { output, .. } => {
                         info!("Using hashed artifact for {:?}", filename);
-                        (output, None)
+                        ZkSolc::handle_output(&output, filename, &mut displayed_warnings)
                     }
-                    None => {
-                        self.prepare_compiler_input(&contract_path).wrap_err(format!(
-                            "Failed to prepare inputs when compiling {:?}",
-                            contract_path
-                        ))?;
+                    CachedContractEntry::Missing { cache } => {
+                        self.prepare_compiler_input(&contract_path, &cache.metadata_path)
```

Comment:
why not pass `base_path` here? so we can also use it directly inside `prepare_compiler_input` instead of getting the `.parent()`

Reply:
Yeah I added that in the later commits, makes sense to replace it